### PR TITLE
Updates to work with changes to REST API

### PR DESCRIPTION
**Pull request by karlrl**
New REST API changes will be landing soon, so this PR will track all the changes necessary to make the python bindings remain compatible. Changes to the Python API should be minimal.

### CODE REVIEW

File: analyzere/base_resources.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def upload_data(self, file_or_str):
 
         # Block until data has finished processing
         while True:
-            resp = self.status
+            resp = self.processing_status
```
Comment (karlrl) [inline analyzere/base_resources.py]: We _could_ rename this field in the API instead.

Comment (pmdarrow) [inline analyzere/base_resources.py]: Is it necessary to change this? We're no more at risk of name collisions than we were before. I think this change could be delayed until we adopt new-style uploads.

**Comment by pmdarrow** [comment]
`accept` and `content-type` change LGTM.

File: analyzere/base_resources.py (hunk 1 above)
Comment (karlrl) [inline analyzere/base_resources.py]: Yes it's necessary—if we don't either stop returning `status` on the root object, or change the name of the property here, the property will get overwritten when the object is retrieved (breaking large-file uploads).

Comment (pmdarrow) [inline analyzere/base_resources.py]: Gotcha.

**Comment by pmdarrow** [comment]
All LGTM.

**Pull request merged**